students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,279 @@
+package com.musala.simple.students;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.maven.wagon.ResourceDoesNotExistException;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+public class Main {
+	private static final String JSON_FILE_EXTENSION = ".json";
+	private static final String ERROR_PATH_MISSING = "Path is missing.";
+	private static final String ERROR_INVALID_PATH = "Invalid path specified.";
+	private static final String ERROR_NOT_A_FILE = "�he path specified is not an existing file.";
+	private static final String ERROR_FILE_NOT_JSON = "The file specified is not of '.json' format.";
+	private static final String ERROR_NO_BACKUP_JSON = "No backup.json exists.";
+	private static final String ERROR_NOT_VALID_JSON = "The content of the JSON file is not a valid JSON string.";
+	private static final String ERROR_STUDENT_NOT_FOUND = "Requested student does not exist in the current Student group.";
+	private static final String BACKUP_JSON_PATH = "src/main/resources/backup.json";
+	private static final String INFO_CREATING_BACKUP = "Creating backup..";
+
+	public static void main(String[] args) {
+
+		if (inputIsValid(args)) {
+
+			String studentsJsonInfo = readFile(args[0]);
+
+			if (!isValidJson(studentsJsonInfo)) {
+				System.err.println(ERROR_NOT_VALID_JSON);
+				try {
+					readBackupJson();
+				} catch (ResourceDoesNotExistException e) {
+					e.printStackTrace();
+				}
+			} else {
+				StudentWrapper studentWrapper = new Gson().fromJson(studentsJsonInfo, StudentWrapper.class);
+				StudentGroup studentGroup = new StudentGroup();
+				studentGroup.fillStudentGroup(studentWrapper.students);
+
+				Map<Integer, Student> studentsMap = studentGroup.getStudents();
+
+				int requestedStudentId = readStudentId(args);
+				boolean studentExists = false;
+				if (requestedStudentId != -1) {
+					try {
+						Student student = studentGroup.getStudentById(requestedStudentId);
+						studentExists = true;
+						printStudentDetails(student);
+					} catch (IllegalArgumentException iae) {
+						System.err.println(ERROR_STUDENT_NOT_FOUND);
+					}
+				}
+				if (!studentExists) {
+					List<Student> studentsList = studentsMap.values().stream().collect(Collectors.toList()); 
```

Comment:
Could you use some simpler Map <-> List conversion, e.g. `List<Student> list = new ArrayList<Student>(studentsMap.values());` :)